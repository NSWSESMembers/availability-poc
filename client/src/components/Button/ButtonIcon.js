import React from 'react';
import PropTypes from 'prop-types';
import { Text, TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';

import styles from './styles';

const ButtonIcon = ({ backgroundColor, text, icon, onPress }) => {
  const containerStyles = [styles.buttonIconContainer];

  if (backgroundColor) {
    containerStyles.push({ backgroundColor });
  }

  return (
    <TouchableOpacity onPress={onPress} style={containerStyles}>
      <Icon size={60} name={icon} color="white" />
      <Text style={styles.buttonIconText}>{text}</Text>
    </TouchableOpacity>
  );
};

ButtonIcon.propTypes = {
  backgroundColor: PropTypes.string,
  text: PropTypes.string.isRequired,
  icon: PropTypes.string,
  onPress: PropTypes.func,
};

export default ButtonIcon;
