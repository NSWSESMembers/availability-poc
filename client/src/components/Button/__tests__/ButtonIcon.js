import React from 'react';
import renderer from 'react-test-renderer';

import ButtonIcon from '../ButtonIcon';

const noop = () => {};

test('renders correctly', () => {
  const tree1 = renderer.create(
    <ButtonIcon
      text="text"
      icon="glass"
      onPress={noop}
    />,
  ).toJSON();
  expect(tree1).toMatchSnapshot();

  const tree2 = renderer.create(
    <ButtonIcon
      text="text"
      icon="glass"
      backgroundColor="red"
      onPress={noop}
    />,
  ).toJSON();
  expect(tree2).toMatchSnapshot();
});
