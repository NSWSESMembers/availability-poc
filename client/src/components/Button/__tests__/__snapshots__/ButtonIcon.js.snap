// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "green",
      "flexGrow": 1,
      "justifyContent": "center",
      "margin": 5,
      "minHeight": 150,
      "minWidth": 150,
      "opacity": 1,
      "paddingLeft": 10,
      "paddingRight": 10,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "white",
          "fontSize": 60,
        },
        undefined,
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 18,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    text
  </Text>
</View>
`;

exports[`renders correctly 2`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "red",
      "flexGrow": 1,
      "justifyContent": "center",
      "margin": 5,
      "minHeight": 150,
      "minWidth": 150,
      "opacity": 1,
      "paddingLeft": 10,
      "paddingRight": 10,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "white",
          "fontSize": 60,
        },
        undefined,
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 18,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    text
  </Text>
</View>
`;
